hardware_interface/source/hyperion.c

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright (C) 2015  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/**
+ * @file    hyperion.c
+ * @author  Trung Tran
+ * @date    2021-06-04
+ */
+#include "hyperion.h"
+
+
+/**
+ * @brief
+ * 		Get the data from a panel and its corresponding channel
+ * @details
+ * 		Temperature, photodiodes, voltage, and current values of a panel can be obtained via this function 
+ *      Refer to Hyperion Detailed Design document
+ * @attention
+ * 		This function is used for config 1, which only includes Port, Port Deployable, Starboard, Startboard Deployable and Zenith
+ * @param panel
+ * 		The panel that we want to retrieve data from 
+ * @param channel
+ * 		voltage channel corresponding to that panel
+ * @param temp
+ * 		The temperature in celcius
+ * @param pd
+ *      
+ * @param voltage
+ *      The voltage in mV
+ * @param current
+ *      The current in mA
+ * @return
+ * 		void
+ */
+void hyperion_config_1_value(config_1_panel_t panel, config_1_channel_type_t channel, float* temp, float* pd, float* voltage, float* current) {
```

Comment:
How this function will be called/used?
It looks like if the channel refers to temperature for example, no value will be set for voltage. Can't we like replace all three with `*param`?

Reply:
I can do that but be cautious that when this function is used, the developer should know what kind of value they are getting.